control/timeresp.py

```diff
@@ -1287,16 +1303,18 @@ def _default_time_vector(sys, N=None, tfinal=None, is_step=True):
         # only need to use default_tfinal if not given; N is ignored.
         if tfinal is None:
             # for discrete time, change from ideal_tfinal if N too large/small
-            N = int(np.clip(ideal_tfinal/sys.dt, N_min_dt, N_max))# [N_min, N_max]
-            tfinal = sys.dt * N
+            # [N_min, N_max]
```

Comment:
Makes sense that it should simulate all the way to tfinal if asked to, I was just not being careful about it. Thanks for doing the legwork!